src/common/blockchain.sh

```diff
@@ -1,3 +1,201 @@
 #!/usr/bin/env bash
 
 # Common IBM blockchain platform functions, e.g. to provision a blockchain service
+
+# shellcheck disable=2086
+
+# shellcheck source=src/common/utils.sh
+source "${SCRIPT_DIR}/common/utils.sh"
+
+function setup_service_constants {
+    region_instance=$(echo "$REGION_ID" | cut -d : -f 2)
+
+    if [ "${region_instance}" = "ys1" ]; then
+        export BLOCKCHAIN_SERVICE_NAME="ibm-blockchain-5-staging"
+        export BLOCKCHAIN_SERVICE_PLAN="ibm-blockchain-plan-v1-ga1-starter-staging"
+    else
+        export BLOCKCHAIN_SERVICE_NAME="ibm-blockchain-5-prod"
+        export BLOCKCHAIN_SERVICE_PLAN="ibm-blockchain-plan-v1-ga1-starter-prod"
+    fi
+
+    export BLOCKCHAIN_SERVICE_KEY=Credentials-1
+}
+
+function authenticate_org {
+    org=$1
+    file="blockchain.json"
+
+    BLOCKCHAIN_NETWORK_ID=$(jq --raw-output ".${org}.network_id" ${file})
+    BLOCKCHAIN_KEY=$(jq --raw-output ".${org}.key" ${file})
+    BLOCKCHAIN_SECRET=$(jq --raw-output ".${org}.secret" ${file})
+    BLOCKCHAIN_URL=$(jq --raw-output ".${org}.url" ${file})
+}
+
+function provision_blockchain {
+    if ! cf service ${BLOCKCHAIN_SERVICE_INSTANCE} > /dev/null 2>&1
+    then
+        cf create-service ${BLOCKCHAIN_SERVICE_NAME} ${BLOCKCHAIN_SERVICE_PLAN} ${BLOCKCHAIN_SERVICE_INSTANCE}
+    fi
+    if ! cf service-key ${BLOCKCHAIN_SERVICE_INSTANCE} ${BLOCKCHAIN_SERVICE_KEY} > /dev/null 2>&1
+    then
+        cf create-service-key ${BLOCKCHAIN_SERVICE_INSTANCE} ${BLOCKCHAIN_SERVICE_KEY}
+    fi
+    cf service-key ${BLOCKCHAIN_SERVICE_INSTANCE} ${BLOCKCHAIN_SERVICE_KEY} | tail -n +2 > blockchain.json
```

Comment:
We should do one more check to see if the service know exist, if it does not exist we should put out an error message saying that could not find or create the service.